lib/gooddata/helpers/global_helpers_params.rb

```diff
@@ -92,15 +92,39 @@ def decode_params(params, options = {})
         # the ${not_a_param} will not be replaced
         if options[:resolve_reference_params]
           regexps = Regexp.union(/\\\\/, /\\\$/, /\$\{(\w+)\}/)
-          data_params.gsub!(regexps) do |match|
-            if match =~ /\\\\/
-              '\\\\'
-            elsif match =~ /\\\$/
-              '$'
-            elsif match =~ /\$\{(\w+)\}/
-              params["#{$1}"] || raise("The gd_encoded_params parameter contains unknow reference #{$1}") # rubocop: disable Style/PerlBackrefs
+          resolve_reference = lambda do |v|
```

Comment:
It would be good for readability if we didn't violate the Metrics/BlockNesting rule. This may help.